homeworks/06-redux-advanced/src/reducer.ts

```diff
@@ -0,0 +1,76 @@
+import { Action, Reducer } from "redux"
+import { Pizza, State } from "./types";
+
+export enum PizzaActionType {
+  LOAD_LIST = "LOAD_LIST",
+  ADD_TO_BASKET = "ADD_TO_BASKET",
+  REMOVE_FROM_BASKET = "REMOVE_FROM_BASKET"
+}
+
+export interface PizzaAction extends Action<PizzaActionType> {
+  pizzaId?: string,
+  pizzas?: Pizza[]
+}
+
+const pizzaComparator = (a: Pizza, b: Pizza) => a.name.localeCompare(b.name)
+
+export const reducer: Reducer<State, PizzaAction> = (state = { pizza: [], basket: [] }, action) => {
+  let newState = state
+  if (action.type === PizzaActionType.LOAD_LIST) {
```

Comment:
Would it better just?
```
if (action.type === PizzaActionType.LOAD_LIST && action.pizzas) {}
```
anyway i would recommend to use switch within reducer as well due to having lots if/else

Reply:
Switch looks bad when cases are long. So, when chaging to switch, I extracted add and remove handling to separate functions.